src/lib.rs

```diff
@@ -256,51 +256,51 @@ impl Check {
         }
     }
 
-    pub fn with_package_selection(&mut self, selection: PackageSelection) -> &mut Self {
+    pub fn set_package_selection(&mut self, selection: PackageSelection) -> &mut Self {
         self.scope.mode = ScopeMode::DenyList(selection);
         self
     }
 
-    pub fn with_packages(&mut self, packages: Vec<String>) -> &mut Self {
+    pub fn set_packages(&mut self, packages: Vec<String>) -> &mut Self {
         self.scope.mode = ScopeMode::AllowList(packages);
         self
     }
 
-    pub fn with_baseline(&mut self, baseline: Rustdoc) -> &mut Self {
+    pub fn set_baseline(&mut self, baseline: Rustdoc) -> &mut Self {
         self.baseline = baseline;
         self
     }
 
-    pub fn with_release_type(&mut self, release_type: ReleaseType) -> &mut Self {
+    pub fn set_release_type(&mut self, release_type: ReleaseType) -> &mut Self {
         self.release_type = Some(release_type);
         self
     }
 
-    pub fn with_only_explicit_features(&mut self) -> &mut Self {
+    pub fn set_only_explicit_features(&mut self) -> &mut Self {
         self.current_feature_config.features_group = rustdoc_gen::FeaturesGroup::None;
         self.baseline_feature_config.features_group = rustdoc_gen::FeaturesGroup::None;
         self
     }
 
-    pub fn with_default_features(&mut self) -> &mut Self {
+    pub fn set_default_features(&mut self) -> &mut Self {
         self.current_feature_config.features_group = rustdoc_gen::FeaturesGroup::Default;
         self.baseline_feature_config.features_group = rustdoc_gen::FeaturesGroup::Default;
         self
     }
 
-    pub fn with_heuristically_included_features(&mut self) -> &mut Self {
+    pub fn set_heuristically_included_features(&mut self) -> &mut Self {
         self.current_feature_config.features_group = rustdoc_gen::FeaturesGroup::Heuristic;
         self.baseline_feature_config.features_group = rustdoc_gen::FeaturesGroup::Heuristic;
         self
     }
 
-    pub fn with_all_features(&mut self) -> &mut Self {
+    pub fn set_all_features(&mut self) -> &mut Self {
         self.current_feature_config.features_group = rustdoc_gen::FeaturesGroup::All;
         self.baseline_feature_config.features_group = rustdoc_gen::FeaturesGroup::All;
         self
     }
 
-    pub fn with_extra_features(
+    pub fn set_extra_features(
```

Comment:
Hmm these new names are more confusing than the old names here.

For example, compare: "check with default features" vs "check set default features." The former sounds like what it does, the latter sounds like we're overriding which features the crate considers default, or something.

Would it be weird to leave the `impl Check` methods as `with_...()`, and make everything else `set_...()`? "Config set" makes sense, "package selection set" makes sense, "check set" does not really make sense.

Reply:
No, that definitely makes sense.  These aren't "setters" in the same sense as the other ones; they don't take an additional parameter.